src/rngs/adapter/reseeding.rs

```diff
@@ -161,11 +166,14 @@ where R: BlockRngCore + SeedableRng,
     /// * `reseeder`: the RNG to use for reseeding.
     pub fn new(rng: R, threshold: u64, reseeder: Rsdr) -> Self {
         assert!(threshold <= ::core::i64::MAX as u64);
+        register_fork_handler();
```

Comment:
You call this for every new reseeding RNG...

Reply:
I thought about this also. We can use `RESEEDING_RNG_FORK_COUNTER` to see if it has already been registered, by setting it to 0 the first time (and something like `u64::MAX` otherwise).

Reply:
Not unless you're careful to avoid `reseed_counter` ever being set to `MAX`. Obviously it shouldn't, but seems more fragile regarding code adjustment, so better to initialise to 0 then update to 1 maybe?

Reply:
I don't think it matters much. But I would like to see someone forking a process `u64::MAX` times :smile:.